### PR TITLE
WIP Save authfile in container config.

### PR DESCRIPTION
**Pull request by neVERberleRfellerER**
This is one possible approach usable as base for https://github.com/containers/libpod/pull/6188 .

### CODE REVIEW

File: libpod/container.go [hunk 1]
```diff
@@ -414,6 +414,8 @@ type ContainerConfig struct {
 
 	// HealthCheckConfig has the health check command and related timings
 	HealthCheckConfig *manifest.Schema2HealthConfig `json:"healthcheck"`
+
+	Authfile string `json:"authfile"`
```
Comment (vrothberg) [inline libpod/container.go:418]: Please add a comment.

File: libpod/define/container_inspect.go [hunk 2]
```diff
@@ -510,6 +510,8 @@ type InspectContainerHostConfig struct {
 	IOMaximumIOps uint64 `json:"IOMaximumIOps"`
 	// IOMaximumBandwidth is Windows-only and not presently implemented.
 	IOMaximumBandwidth uint64 `json:"IOMaximumBandwidth"`
+	//Authfile is path to used authfile
```
Comment (vrothberg) [inline libpod/define/container_inspect.go:513]:
```suggestion
	// Authfile is the path to the auth file used at container creation.
```

File: pkg/specgen/specgen.go [hunk 3]
```diff
@@ -130,6 +130,8 @@ type ContainerBasicConfig struct {
 	// Remove indicates if the container should be removed once it has been started
 	// and exits
 	Remove bool `json:"remove"`
+	//Authfile is path to authfile to use
```
Comment (vrothberg) [inline pkg/specgen/specgen.go:133]:
```suggestion
	// Authfile is the path to auth file to use
```

**Comment by vrothberg** [comment]
@mheon PTAL Cool to put that in the config or shall we use a label?

**Comment by vrothberg** [comment]
Note that it's important to only store the auth file if it was actually specified by the user on the CLI via `--authfile`. We shouldn't store the default one. This way we are making sure that we can establish a priority for which authfile is being used during auto updates.

**Comment by mheon** [comment]
@vrothberg I'll admit I haven't been paying attention to the previous discussion here - what's the logic for making this a part of container config? Pros/cons vs. using labels? My initial instinct is that we should use a label, but I could be convinced otherwise.

**Comment by vrothberg** [comment]
> @vrothberg I'll admit I haven't been paying attention to the previous discussion here - what's the logic for making this a part of container config? Pros/cons vs. using labels? My initial instinct is that we should use a label, but I could be convinced otherwise.

@neVERberleRfellerER is currently working on wiring authfile support into `podman auto-update`. A `podman auto-update --authfile=...` is already a big step forward but it's not quite enough for @neVERberleRfellerER's use case. Different containers might be pulled from the same registry but from different namespaces which may require different credentials. Hence, one authfile isn't enough.

The idea now is to store the authfile at container-creation time. During auto-update, we can look up if a container was created with a non-default authfile and use that instead when contacting the registry.

I recall that you preferred storing certain data in labels rather than extending the container config. This may be such a case.

**Comment by mheon** [comment]
My initial impression from this is that it should probably be a label similar to the ones we use for Inspect options (it's not strictly necessary for Libpod operation, but if it's there we'll know what to do with it).

**Comment by neVERberleRfellerER** [comment]
Since https://github.com/containers/libpod/pull/6188 has been merged this isn't useful anymore. Let me know if you want authfile in container config for different reasons.
@vrothberg Thank you for everything.

**Pull request closed without merging**